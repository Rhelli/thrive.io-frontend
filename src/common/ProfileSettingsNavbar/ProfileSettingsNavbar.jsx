import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';
import tLogo from '../../assets/img/thrive-t-transparent.png';
import styles from './ProfileSettingsNavbar.module.scss';

const ProfileSettingsNavbar = ({ handleBackButtonClick }) => (
  <div className={styles.profileSettingsNavbarContainer}>
    <span className={styles.titleSpan}>
      <button type="button" onClick={handleBackButtonClick} onKeyUp={handleBackButtonClick}>
        <FontAwesomeIcon icon={faChevronLeft} />
      </button>
      <h1>Settings</h1>
    </span>
    <span className={styles.imageSpan}>
      <img src={tLogo} alt="Thrive" />
    </span>
  </div>
);

ProfileSettingsNavbar.propTypes = {
  handleBackButtonClick: PropTypes.func.isRequired,
};

export default ProfileSettingsNavbar;
