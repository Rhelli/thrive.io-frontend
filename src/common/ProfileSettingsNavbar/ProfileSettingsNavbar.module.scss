.profileSettingsNavbarContainer {
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 50px;
  padding: 5px 0;
  border: 1px solid green;
}

.profileSettingsNavbarBackButton {
  height: 100%;
  width: 40px;

  button {
    height: 100%;
    width: 100%;
    outline: 0;
    border: 0;
    background-color: transparent;

    svg {
      font-size: 18px;
    }
  }
}

.profileSettingsNavbarTitle {
  margin: 0 auto;
}

.profileSettingsNavbarUser {
  width: 40px;
  height: 100%;
  
  div {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 100%;
    border-radius: 50%;
    border: 2px solid grey;

    svg {
      font-size: 18px;
    }
  }
}