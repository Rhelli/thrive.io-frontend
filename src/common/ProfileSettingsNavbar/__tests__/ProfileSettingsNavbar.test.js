import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import ProfileSettingsNavbar from '../ProfileSettingsNavbar';

const handleBackButtonClick = jest.fn(() => 'Clicked!');

const profileSettingsNavbar = (
  <ProfileSettingsNavbar handleBackButtonClick={handleBackButtonClick} />
);

describe('The appearance and functionality of the ProfileSettingsNavbar component', () => {
  it('Should match the previous snapshot', () => {
    const tree = renderer.create(profileSettingsNavbar).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has a master container with the correct attributes', () => {
    const { container } = render(profileSettingsNavbar);
    expect(container.firstChild.nodeName).toBe('DIV');
    expect(container.firstChild.classList.contains('profileSettingsNavbarContainer')).toBe(true);
  });

  it('Contains the title span element with the button and title inside it', () => {
    const { container, getByText } = render(profileSettingsNavbar);
    const span = container.firstChild.firstChild;
    expect(span.nodeName).toBe('SPAN');
    expect(span.firstChild.nodeName).toBe('BUTTON');
    expect(span.firstChild.firstChild.nodeName).toBe('svg');
    expect(span.lastChild.nodeName).toBe('H1');
    getByText('Settings');
  });

  it('Contains the image span with the thrive logo inside of it', () => {
    const { container } = render(profileSettingsNavbar);
    const span = container.firstChild.lastChild;
    expect(span.firstChild.nodeName).toBe('IMG');
    expect(span.firstChild).toHaveAttribute('src', 'thrive-t-transparent.png');
  });
});
