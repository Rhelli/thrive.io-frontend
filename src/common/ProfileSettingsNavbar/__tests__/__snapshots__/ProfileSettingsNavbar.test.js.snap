// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the ProfileSettingsNavbar component Should match the previous snapshot 1`] = `
<div
  className="profileSettingsNavbarContainer"
>
  <span
    className="titleSpan"
  >
    <button
      onClick={[MockFunction]}
      onKeyUp={[MockFunction]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-left fa-w-10 "
        data-icon="chevron-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <h1>
      Settings
    </h1>
  </span>
  <span
    className="imageSpan"
  >
    <img
      alt="Thrive"
      src="thrive-t-transparent.png"
    />
  </span>
</div>
`;
